setup.py

```diff
@@ -1,308 +1,123 @@
-
-# Notes regarding coverage testing.
-#
-# The --gcov command line option configures GCC to collect code coverage data
-# for testing purposes. The raw collected data can be converted to .gcov files
-# with the command "gcov build/temp.linux-x86_64-3.4/src/gmpy2.gcno". To create
-# a tidy HTML summary, use "gcovr -s -r . --html -o coverage.html". When tests
-# are completed, delete the .gcov and coverage.html files.
-#
-# The following commands show the use of lcov to analyze the coverage data.
-#
-#   py34 setup.py clean
-#   py34 setup.py install --gcov --force
-#   py34 test/runtests.py
-#   lcov --capture --directory build/temp.linux-x86_64-3.4/src/gmpy2.gcno --output-file build/coverage.info
-#   genhtml build/coverage.info --output-directory build/out
-#   firefox build/out/index.html
-#
-# Remember to remove the *.gcov file and the out sub-directory.
-
-defines = []
-build_type_specified = False
-
-for token in sys.argv[:]:
-    if token.lower() == '--force':
-        defines.append( ('FORCE', 1) )
-        sys.argv.remove(token)
-
-    if token.lower() == '--lib64':
-        lib_path = 'lib64'
-        sys.argv.remove(token)
-
-    if token.lower() == '--lib32':
-        lib_path = 'lib32'
-        sys.argv.remove(token)
-
-    if token.lower() == '--vector':
-        defines.append( ('VECTOR', 1) )
-        sys.argv.remove(token)
-
-    if token.lower() == '--fast':
-        defines.append( ('FAST', 1) )
-        sys.argv.remove(token)
-
-    if token.lower() == '--msys2':
-        defines.append( ('MSYS2', 1) )
-        sys.argv.remove(token)
-
-    if token.lower() == '--mpir':
-        defines.append( ('MPIR', 1) )
-        sys.argv.remove(token)
-
-    if token.lower().startswith('--shared'):
-        build_type_specified = True
-        try:
-            defines.append( ('SHARED', token.split('=')[1]) )
-        except IndexError:
-            defines.append( ('SHARED', 1) )
-        sys.argv.remove(token)
-
-    if token.lower().startswith('--static'):
-        build_type_specified = True
-        try:
-            defines.append( ('STATIC', token.split('=')[1]) )
-        except IndexError:
-            defines.append( ('STATIC', 1) )
-        sys.argv.remove(token)
-
-    if token.lower() == '--gcov':
-        defines.append( ('GCOV', 1) )
-        sys.argv.remove(token)
-
-if not build_type_specified:
-        defines.append( ('SHARED', 1) )
-
-
-# Improved clean command.
-
-class gmpy_clean(clean):
-
-    def run(self):
-        self.all = True
-        clean.run(self)
-
-# Define a custom build class that parses to the defined macros to alter the
-# build setup.
-
-class gmpy_build_ext(build_ext):
+import platform
+from setuptools import setup, find_packages, Extension
+from setuptools.command.build_ext import build_ext
+
+ON_WINDOWS = platform.system() == 'Windows'
+_comp_args = ["DSHARED=1"]
+link_args = []
+
+sources = ['src/gmpy2.c']
+_libs = ['mpfr', 'mpc'] if not ON_WINDOWS else [
+    'libmpfr', 'libmpc']
```

Comment:
`lib` should not be prepended to the library name even on windows

Reply:
Yes, it should. MSVC uses `libmpfr` to refer to the library called `mpfr` in posix. See similar code [in this project](https://github.com/ParallelSSH/ssh2-python/blob/master/setup.py#L26-L30) which has working windows wheels.

Reply:
gmpy2 has working windows wheels and it this isn't used. the project you linked builds `libssh2.lib` even in windows. MSVC build of mpfr builds `mpfr.lib`, so this will not work.

Reply:
You're right, found these other mpir builds that also call it `mpir.lib`. Todo.